src/main/java/net/snowflake/ingest/streaming/internal/ClientBufferParameters.java

```diff
@@ -124,4 +124,8 @@ public Optional<Integer> getMaxRowGroups() {
   public String getParquetMessageTypeName() {
     return isIcebergMode ? PARQUET_MESSAGE_TYPE_NAME : BDEC_PARQUET_MESSAGE_TYPE_NAME;
   }
+
+  public boolean isEnableDictionaryEncoding() {
+    return isIcebergMode;
```

Comment:
this *might* be dependent on storage serialization policy too, lets verify. no need to hold up PR.